### PR TITLE
#304 Fix product attributes

### PR DESCRIPTION
**Pull request by TomRadford**

### CODE REVIEW

File: src/components/Product/Attribute.svelte [hunk 1]
```diff
@@ -5,48 +5,55 @@
...
+  const onDeleteClick = () => {
     deleteAttribute(attribute);
   };
 
-  $: {
-    if (attribute.isToggle()) {
-      attribute.meta.value = attribute.toggleValue ? 'Yes' : 'No';
-    }
-  }
+  $: attributeIndex = $product.workingCopy.meta.findIndex((att) => att.key === attribute.meta?.key);
```
Comment (TomRadford) [inline src/components/Product/Attribute.svelte:21]: @matthampton-stock2shop looks like your IDE isnt looking at our prettier config sadly

File: src/models/version-controlled.ts [hunk 2]
```diff
@@ -60,95 +58,91 @@ export default class ObjectVersionControl<T> {
...
     }
-    return left === right
+    return left === right;
   }
 
   areArraysEqual(keyPath: string[], left, right) {
     if (left.length !== right.length) {
-      console.info("Changed key", {keyPath, left, right})
-      return false
+      console.info('Changed key', { keyPath, left, right });
```
Comment (TomRadford) [inline src/models/version-controlled.ts:131]: @matthampton-stock2shop seems like WebStorm doesnt seem to be formatting on our prettier config

Comment (matthampton-stock2shop) [reply src/models/version-controlled.ts:131]: Ah ... Maybe I need to ask it to autoformat - I haven't been in the habit of using that feature

**Comment by TomRadford** [comment]
@matthampton-stock2shop rendering of input if blank value should be resolved now

**Review by Ayano2000 — CHANGES_REQUESTED** [review]
I have tested basic `CRUD` for `productmeta`, adding and removing meta via templates (removing all meta items for a template does deselect the template).

- If a template is removed, any meta that it may have added to a product stays

- [ ]  Default `metatemplate` values do not show when adding a template

-------------

I believe this is not a part of this PR but:
- I cannot edit any of the existing meta_templates, `Successfully added new however some attribute(s) had an error. Error: Undefined property: stdClass::$name`
- Sometimes the modal for creating/editing meta templates gets cached.
- When adding a `metatemplate` of type dropdown, we cannot add any selectable values. 

@TomRadford I'm not sure all of these issues are related to this PR, but I think showing a templates default values when adding it should be working before we merge this

**Review by Ayano2000 — APPROVED** [review]
LGTM

**Pull request merged**